drivers/charger/charger_max20335.c

```diff
@@ -246,14 +306,136 @@ static int max20335_set_prop(const struct device *dev, charger_prop_t prop,
 	}
 }
 
+static int max20335_enable_interrupt_pin(const struct device *dev, bool enabled)
+{
+	const struct charger_max20335_config *const config = dev->config;
+	gpio_flags_t flags;
+	int ret;
+
+	flags = enabled ? GPIO_INT_LEVEL_ACTIVE : GPIO_INT_DISABLE;
+
+	ret = gpio_pin_interrupt_configure_dt(&config->int_gpio, flags);
+	if (ret < 0) {
+		LOG_ERR("Could not %s interrupt GPIO callback: %d", enabled ? "enable" : "disable",
+			ret);
+	}
+
+	return ret;
+}
+
+static void gpio_callback(const struct device *dev, struct gpio_callback *cb, uint32_t pins)
+{
+	struct charger_max20335_data *data = CONTAINER_OF(cb, struct charger_max20335_data,
+							  gpio_cb);
+	int ret;
+
+	(void) max20335_enable_interrupt_pin(data->dev, false);
+
+	ret = k_work_submit(&data->int_routine_work);
+	if (ret < 0) {
+		LOG_WRN("Could not submit int work: %d", ret);
+	}
+}
+
+static void int_routine_work_handler(struct k_work *work)
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/pull/69465/commits/2ee5b6fd8045446ef24fe04666deb140d3ae3f5e#r1507824755  similar suggestion applies here

Reply:
Fixed here and also for `int_enable_work_handler`.